firmware/thruna/PDM/src/app/app_powerManager.c

```diff
@@ -1,49 +1,131 @@
 #include "app_powerManager.h"
 #include "app_canRx.h"
+#include "app_retryHandler.h"
+#include "app_canTx.h"
+#include <stdio.h>
 
 typedef struct
 {
-    bool efuses[NUM_EFUSE_CHANNELS];
+    RetryConfig retry_configs[NUM_EFUSE_CHANNELS];
 } PowerStateConfig;
 
+static RetryData retry_data[NUM_EFUSE_CHANNELS];
+
 static const PowerStateConfig power_states_config[NUM_POWER_STATES] = {
     [POWER_MANAGER_SHUTDOWN] = {
-        .efuses = {
-            [EFUSE_CHANNEL_AIR] = true,
-            [EFUSE_CHANNEL_LVPWR] = true,
-            [EFUSE_CHANNEL_EMETER] = false,
-            [EFUSE_CHANNEL_AUX] = false,
-            [EFUSE_CHANNEL_DRS] = false,
-            [EFUSE_CHANNEL_FAN] = false,
-            [EFUSE_CHANNEL_DI_LHS] = false,
-            [EFUSE_CHANNEL_DI_RHS] = false,
+        .retry_configs = {
+            [EFUSE_CHANNEL_AIR] = {true, RETRY_PROTOCOL_AIR, 1},
+            [EFUSE_CHANNEL_LVPWR] = {true, RETRY_PROTOCOL_LV, 1},
+            [EFUSE_CHANNEL_EMETER] = {false, RETRY_PROTOCOL_EMETER, 3},
+            [EFUSE_CHANNEL_AUX] = {false, RETRY_PROTOCOL_AUX, 3},
+            [EFUSE_CHANNEL_DRS] = {false, RETRY_PROTOCOL_DRS, 3},
+            [EFUSE_CHANNEL_FAN] = {false, RETRY_PROTOCOL_FANS, 3},
+            [EFUSE_CHANNEL_DI_LHS] = {false, RETRY_PROTOCOL_DI_LHS, 1},
+            [EFUSE_CHANNEL_DI_RHS] = {false, RETRY_PROTOCOL_DI_RHS, 1},
         }
     },
     [POWER_MANAGER_DRIVE] = {
-        .efuses = {
-            [EFUSE_CHANNEL_AIR] = true,
-            [EFUSE_CHANNEL_LVPWR] = true,
-            [EFUSE_CHANNEL_EMETER] = false,
-            [EFUSE_CHANNEL_AUX] = false,
-            [EFUSE_CHANNEL_DRS] = false,
-            [EFUSE_CHANNEL_FAN] = true,
-            [EFUSE_CHANNEL_DI_LHS] = true,
-            [EFUSE_CHANNEL_DI_RHS] = true,
+        .retry_configs = {
+            [EFUSE_CHANNEL_AIR] = {true, RETRY_PROTOCOL_AIR, 1},
+            [EFUSE_CHANNEL_LVPWR] = {true, RETRY_PROTOCOL_LV, 1},
+            [EFUSE_CHANNEL_EMETER] = {false, RETRY_PROTOCOL_EMETER, 3},
+            [EFUSE_CHANNEL_AUX] = {false, RETRY_PROTOCOL_AUX, 3},
+            [EFUSE_CHANNEL_DRS] = {false, RETRY_PROTOCOL_DRS, 3},
+            [EFUSE_CHANNEL_FAN] = {true, RETRY_PROTOCOL_FANS, 3},
+            [EFUSE_CHANNEL_DI_LHS] = {true, RETRY_PROTOCOL_DI_LHS, 1},
+            [EFUSE_CHANNEL_DI_RHS] = {true, RETRY_PROTOCOL_DI_RHS, 1},
         }
     }
 };
 
+void app_powerManager_init()
+{
+    for (int efuse = 0; efuse < NUM_EFUSE_CHANNELS; efuse++)
+    {
+        retry_data[efuse].retry_state          = RETRY_STATE_OFF;
+        retry_data[efuse].retry_attempts       = 0;
+        retry_data[efuse].timer_attempts       = 0;
+        retry_data[efuse].timer_attempts_limit = 3;
```

Comment:
I think it'd be nice for this to be a config parameter too. 
(If I understand correctly, this is the number of cycles over which you average the current?)

Also, I think this should be a lot higher, I believe this value is 3 x 10ms = 30ms. We want on the order of a few seconds to be certain.

Reply:
I made `timer_attempts_limit = 200` so 200 * 10ms = 2s is that enough time?